lib/puppet/functions/katello/certificate_subject.rb

```diff
@@ -0,0 +1,24 @@
+# @summary
+#   Extracts a certificate's subject in RFC2253 form
+#
+# @example How to extract a certificate's subject
+#   $cert_subject = katello::certificate_subject($path_to_certificate)
+#
+Puppet::Functions.create_function(:'katello::certificate_subject') do
+  # @param certificate_path
+  # @param options
+  #
+  # @return [String]
+  dispatch :certificate_subject do
+    param 'Any', :certificate_path
```

Comment:
Should this be an absolute path?
```suggestion
    param 'Stdlib::Absolutepath', :certificate_path
```

Reply:
Oh, I now read the note:
>  Deferred functions can only use types that are built into Puppet (for example String). They cannot use types from modules like stdlib because Puppet does not plugin-sync these types to the agent. 

Perhaps it's better to use a non-empty string instead?
```suggestion
    param 'String[1]', :certificate_path
```